### DISCUSSION TITLE
Progress and transition

### DISCUSSION THREAD

**Discussion by YpsilonTM** [saadeghi/daisyui#2209] [Q&A] [answered]
I want the progress bar to go smooth when the value changes but I have tried a few things but I can't seem to find how to get smooth transitions with the progress bar. Is this even possible?
`<progress class="progress transition-all" {value} max="100" />`

 I found [this ](https://stackoverflow.com/questions/24408442/css3-transition-of-html5-progress-bar-element) but that doesn't even work for me. Is there a way that anyone knows of or is it impossible.

**Comment by YpsilonTM** [comment, marked as answer]
My apologies I didn't dive deep enough into the web. I have found the solution myself.
`[&::-webkit-progress-value]:transition-all`

**Reply by rafalkrupinski** [reply]
I don't think you should have to use a prefix.

**Reply by YpsilonTM** [reply]
You mean just `transition-all` ? I have tried that it doens't work.

**Reply by saadeghi** [reply]
Webkit and Mozilla have confusing names for progress selector.
- [`::-webkit-progress-bar`](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-progress-bar) is for the background of the progress
- [`::-webkit-progress-value`](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-progress-value) is for the foreground of the progress
- [`::-moz-progress-bar`](https://developer.mozilla.org/en-US/docs/Web/CSS/::-moz-progress-bar) unlike `::-webkit-progress-bar` is for the **foreground** of the progress

So here you should use both `::-webkit-progress-value` and `::-moz-progress-bar` to style the value bar. As far as I know, even Autoprefixer doesn't help here and you should include prefixes yourself.

**Reply by condemil** [reply]
Looks like `[&::-moz-progress-bar]:transition-all` doesn't work in Firefox, I tried version 122 on Linux.

**Reply by loranger** [reply]
Maybe I'm doing wrong but unfortunately thins does not seems to work using chrome or firefox
```html
<progress
        x-ref="progress"
        class="progress [&::-webkit-progress-value]:transition-all [&::-moz-progress-bar]:transition-all block"
        :class="isPlaying ? 'progress-secondary' : 'progress-warning'"
        value="0"
        max="100"></progress>
```
I do update my value using alpinejs
```javascript
this.$refs.audio.addEventListener('timeupdate', () => {
    var percent = Math.round(this.$refs.audio.currentTime * 100 / this.$refs.audio.duration);
    this.$refs.progress.value = percent;
})
```

How do you guys update smoothly your progress bars?
